### DISCUSSION TITLE
Adding InWater and NotInWater as possible conditionnal

### DISCUSSION THREAD

**Discussion by Foxtrot39** [FakeFishGames/Barotrauma#7224] [Ideas and suggestions]
Currently InWater and NotInWater are only for SE type wich make them impossible to combine with other SE without doing gymnastic using another SE to give a tag to then use said tag as conditional to trigger either the under water effect or the in dry area effect.

Ex : a grenade deal internal damage due to water pressure wave when detonated under wtaer but spawn schrapnel when in a dry area

Both use type="OnBroken" to trigger the explosion and remove the item afterwards, to trigger either of the two SE you will need to make 2 other SE, one with type="InWater"giving a "InWater" tag,another with type="NotInWater" for "NotInWater" tag and use the tag as a conditional to trigger their respective SE.

However the more you have to add like this the more it is likely to break in the end, as I discovered while giving a static object (machine) a tag whenever its underwtaer or not and a SE depending on wich tag is active only for it to either totaly ignore the one or the other.

Playing with tag duration doesn't really solve the issue as only one of the two seems to ever trigger.

**Comment by Regalis11** [comment]
Both items and characters have a property called "InWater" which can be checked using conditionals.

```
<Conditional InWater="true" />
<Conditional InWater="false" />
```

**Reply by Foxtrot39** [reply]
Oh, that's how its used as conditional. Thanks a lot, this allow me to do so much more with my mod
